From 41b89ee7febac04fa241acf59097cfa8d7835087 Mon Sep 17 00:00:00 2001
From: Daniel Nicoletti <dantti85-pk@yahoo.com.br>
Date: Wed, 3 Nov 2010 17:47:17 -0200
Subject: [PATCH] aptcc: Set the env proxy vars so that Apt::Acquire is not overwritten, fixes LP: #633008

---
 backends/aptcc/apt.cpp |   18 ++++++------------
 1 files changed, 6 insertions(+), 12 deletions(-)

diff --git a/backends/aptcc/apt.cpp b/backends/aptcc/apt.cpp
index c4089de..74c6dd0 100644
--- a/backends/aptcc/apt.cpp
+++ b/backends/aptcc/apt.cpp
@@ -66,8 +66,8 @@ aptcc::aptcc(PkBackend *backend, bool &cancel)
 bool aptcc::init()
 {
 	gchar *locale;
-	gchar *proxy_http;
-	gchar *proxy_ftp;
+	gchar *http_proxy;
+	gchar *ftp_proxy;
 
 	// Set PackageKit status
 	pk_backend_set_status(m_backend, PK_STATUS_ENUM_LOADING_CACHE);
@@ -84,18 +84,12 @@ bool aptcc::init()
 	}
 
 	// set http proxy
-	if (proxy_http = pk_backend_get_proxy_http(m_backend)) {
-		_config->Set("Acquire::http::Proxy", proxy_http);
-	} else {
-		_config->Set("Acquire::http::Proxy", "");
-	}
+	http_proxy = pk_backend_get_proxy_http(m_backend);
+	setenv("http_proxy", http_proxy, 1);
 
 	// set ftp proxy
-	if (proxy_ftp = pk_backend_get_proxy_ftp(m_backend)) {
-		_config->Set("Acquire::ftp::Proxy", proxy_ftp);
-	} else {
-		_config->Set("Acquire::ftp::Proxy", "");
-	}
+	ftp_proxy = pk_backend_get_proxy_ftp(m_backend);
+	setenv("ftp_proxy", ftp_proxy, 1);
 
 	packageSourceList = new pkgSourceList;
 	// Read the source list
-- 
1.6.1

