From ce1b5311d1663935a4fff8789ebf9b7f0b728145 Mon Sep 17 00:00:00 2001
From: Matthias Klumpp <matthias@tenstral.net>
Date: Wed, 9 Apr 2014 23:12:11 +0200
Subject: [PATCH] aptcc: Update default changelog URL

---
 backends/aptcc/apt-utils.cpp |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/backends/aptcc/apt-utils.cpp b/backends/aptcc/apt-utils.cpp
index b774c29..9c810f5 100644
--- a/backends/aptcc/apt-utils.cpp
+++ b/backends/aptcc/apt-utils.cpp
@@ -191,7 +191,7 @@ bool downloadChangelog(AptCacheFile &CacheFile,
                        string targetfile)
 /* Download a changelog file for the given package version to
  * targetfile. This will first try the server from Apt::Changelogs::Server
- * (http://packages.debian.org/changelogs by default) and if that gives
+ * (http://metadata.ftp-master.debian.org/changelogs by default) and if that gives
  * a 404 tries to get it from the archive directly (see
  * GuessThirdPartyChangelogUri for details how)
  */
@@ -212,7 +212,7 @@ bool downloadChangelog(AptCacheFile &CacheFile,
 
    // make the server root configurable
    server = _config->Find("Apt::Changelogs::Server",
-                          "http://packages.debian.org/changelogs");
+                          "http://metadata.ftp-master.debian.org/changelogs");
    path = GetChangelogPath(CacheFile, Pkg, Ver);
 
    if (origin.compare("Ubuntu") == 0)
-- 
1.7.1

