From 273c64848ce18b0bf498ec8bcb5c43d54a5c6c5b Mon Sep 17 00:00:00 2001
From: Matthias Klumpp <matthias@tenstral.net>
Date: Tue, 22 Jul 2014 12:28:59 +0200
Subject: [PATCH] Make browser-plugin search for npapi-sdk or mozilla-plugin

---
 configure.ac |    8 +++++++-
 1 files changed, 7 insertions(+), 1 deletions(-)

--- a/configure.ac
+++ b/configure.ac
@@ -318,7 +318,7 @@
 	      enable_browser_plugin=$enableval,enable_browser_plugin=yes)
 if test x$enable_browser_plugin = xyes; then
 	PKG_CHECK_MODULES(PK_BROWSER_PLUGIN,	\
-			  mozilla-plugin >= 8.0	\
+			  mozilla-plugin	\
 			  gio-unix-2.0		\
 			  nspr >= 4.8		\
 			  cairo			\
