From 3acb508b95e737eadcfaca02f9aaa88ab48a6e87 Mon Sep 17 00:00:00 2001
From: Richard Hughes <richard@hughsie.com>
Date: Tue, 23 Sep 2014 14:54:19 +0100
Subject: [PATCH] Fix compile when the systemd headers are not located in
 /usr/include

---
 client/Makefile.am | 1 +
 1 file changed, 1 insertion(+)

diff --git a/client/Makefile.am b/client/Makefile.am
index 451f47f..d9705ea 100644
--- a/client/Makefile.am
+++ b/client/Makefile.am
@@ -25,6 +25,7 @@ AM_CPPFLAGS =						\
 	$(GIO_CFLAGS)					\
 	$(DBUS_CFLAGS)					\
 	$(SQLITE_CFLAGS)				\
+	$(SYSTEMD_CFLAGS)				\
 	-DBINDIR=\"$(bindir)\"			 	\
 	-DDATADIR=\"$(datadir)\"			\
 	-DPREFIX=\""$(prefix)"\" 			\
