From c3eed50835846b0357693cceb1d4654713c30e94 Mon Sep 17 00:00:00 2001
From: Daniel Nicoletti <dantti12@gmail.com>
Date: Mon, 10 Dec 2012 15:56:23 -0200
Subject: [PATCH] aptcc: Fix bug that was marking some packages as auto-installed and confused the resolver

---
 backends/aptcc/apt-intf.cpp |   10 +---------
 1 files changed, 1 insertions(+), 9 deletions(-)

--- a/backends/aptcc/apt-intf.cpp
+++ b/backends/aptcc/apt-intf.cpp
@@ -1475,7 +1475,6 @@
     Cache.MarkDelete(Pkg, false);
 }
 
-
 bool AptIntf::tryToInstall(const pkgCache::VerIterator &ver,
                            pkgDepCache &Cache,
                            pkgProblemResolver &Fix,
@@ -1507,14 +1506,6 @@
         ExpectedInst++;
     }
 
-    // 	cout << "trytoinstall ExpectedInst " << ExpectedInst << endl;
-    // Install it with autoinstalling enabled (if we not respect the minial
-    // required deps or the policy)
-    if ((State.InstBroken() == true || State.InstPolicyBroken() == true) &&
-            BrokenFix == false) {
-        Cache.MarkInstall(Pkg,true);
-    }
-
     return true;
 }
 
