Index: packagekit-0.6.10/backends/aptcc/apt.cpp
===================================================================
--- packagekit-0.6.10.orig/backends/aptcc/apt.cpp	2010-12-20 17:20:47.975940660 +0000
+++ packagekit-0.6.10/backends/aptcc/apt.cpp	2010-12-20 17:20:50.667940659 +0000
@@ -262,7 +262,7 @@
 	}
 
 	if (filters != 0) {
-		std::string str = ver.Section();
+		std::string str = ver.Section() == NULL ? "" : ver.Section();
 		std::string section, repo_section;
 
 		size_t found;
@@ -554,7 +554,7 @@
 void aptcc::emit_details(const pkgCache::PkgIterator &pkg)
 {
 	pkgCache::VerIterator ver = find_ver(pkg);
-	std::string section = ver.Section();
+	std::string section = ver.Section() == NULL ? "" : ver.Section();
 
 	size_t found;
 	found = section.find_last_of("/");
@@ -610,7 +610,7 @@
     if (origin.compare("Debian") == 0 || origin.compare("Ubuntu") == 0) {
         string prefix;
 
-        string src_section = candver.Section();
+        string src_section = candver.Section() == NULL ? "" : candver.Section();
         if(src_section.find('/') != src_section.npos) {
             src_section = string(src_section, 0, src_section.find('/'));
         } else {
Index: packagekit-0.6.10/backends/aptcc/pk-backend-aptcc.cpp
===================================================================
--- packagekit-0.6.10.orig/backends/aptcc/pk-backend-aptcc.cpp	2010-12-20 17:30:05.147940660 +0000
+++ packagekit-0.6.10/backends/aptcc/pk-backend-aptcc.cpp	2010-12-20 17:30:12.043940661 +0000
@@ -967,7 +967,7 @@
 		// Ignore virtual packages
 		pkgCache::VerIterator ver = m_apt->find_ver(pkg);
 		if (ver.end() == false) {
-			string section = pkg.VersionList().Section();
+			string section = pkg.VersionList().Section() == NULL ? "" : pkg.VersionList().Section();
 
 			size_t found;
 			found = section.find_last_of("/");
