diff -up PackageKit-0.6.18/src/pk-main.c.signal PackageKit-0.6.18/src/pk-main.c
--- PackageKit-0.6.18/src/pk-main.c.signal	2011-09-08 20:17:51.479976840 -0400
+++ PackageKit-0.6.18/src/pk-main.c	2011-09-08 20:20:33.134971062 -0400
@@ -271,13 +271,13 @@ main (int argc, char *argv[])
 		goto exit_program;
 	}
 
-#if GLIB_CHECK_VERSION(2,29,4)
+#if GLIB_CHECK_VERSION(2,29,90)
 	/* do stuff on ctrl-c */
-	g_unix_signal_add_watch_full (SIGINT,
-				      G_PRIORITY_DEFAULT,
-				      pk_main_sigint_cb,
-				      loop,
-				      NULL);
+	g_unix_signal_add_full (G_PRIORITY_DEFAULT,
+                                SIGINT,
+			        pk_main_sigint_cb,
+			        loop,
+			        NULL);
 #else
 	signal (SIGINT, pk_main_sigint_handler);
 #endif
