commit bccc8547f7187fbf4837f2ec015f0ab148df2020
Author: Matthias Klumpp <matthias@tenstral.net>
Date:   Sat Jun 14 12:27:35 2014 +0200

    Automatically fix permissions of transactions.db

diff --git a/src/pk-transaction-db.c b/src/pk-transaction-db.c
index c20164d..2bb52ab 100644
--- a/src/pk-transaction-db.c
+++ b/src/pk-transaction-db.c
@@ -34,6 +34,7 @@
 #endif /* HAVE_UNISTD_H */
 
 #include <glib/gi18n.h>
+#include <glib/gstdio.h>
 #include <sqlite3.h>
 #include <packagekit-glib2/pk-enum.h>
 #include <packagekit-glib2/pk-results.h>
@@ -979,6 +980,9 @@ pk_transaction_db_load (PkTransactionDb *tdb, GError **error)
 			goto out;
 	}
 
+	/* try to set correct permissions */
+	g_chmod (PK_DB_DIR "/transactions.db", 0644);
+
 	/* success */
 	tdb->priv->loaded = TRUE;
 	ret = TRUE;
