Description: Set PackageKit default config
Forwarded: not-needed
Author: Matthias Klumpp <mak@debian.org>
Last-Update: 2013-12-04
--- a/etc/PackageKit.conf.in
+++ b/etc/PackageKit.conf.in
@@ -96,16 +96,16 @@
 # cause errors if the desktop client is trying to resolve a large number of
 # packages in one method.
 #
-# default=1200
-MaximumItemsToResolve=1200
+# default=4800
+MaximumItemsToResolve=4800

 # The maximum number of packages that can be processed in one method
 #
 # Setting this lower decreases the risk of a local denial of service, but may
 # cause errors if the desktop client is trying to do a large transaction.
 #
-# default=5000
-MaximumPackagesToProcess=5000
+# default=5200
+MaximumPackagesToProcess=5200

 # How long the transaction is valid before it's destroyed, in seconds
 #
