Description: Set PackageKit default config
Forwarded: not-needed
Author: Matthias Klumpp <matthias@nlinux.org>
Last-Update: 2011-07-14
--- a/etc/PackageKit.conf.in
+++ b/etc/PackageKit.conf.in
@@ -219,7 +219,7 @@
 # cause errors if the desktop client is trying to do a large transaction.
 #
 # default=2500
-MaximumPackagesToProcess=2500
+MaximumPackagesToProcess=4000
 
 # How long the transaction is valid before it's destroyed, in seconds
 #
@@ -275,7 +275,7 @@
 # Seporate entries can be seporated with the ';' character.
 #
 # default=/usr/lib*/*/firefox
-NoUpdateProcessList=
+NoUpdateProcessList=/usr/lib/*/firefox;/usr/lib/*/iceweasel
 
 # If the backend should use threads to avoid blocking the daemon for other users
 #
