Set Debian PolKit policy for PackageKit
Forwarded: not-needed
Author: Matthias Klumpp <matthias@tenstral.net>
Last-Update: 2010-12-06
--- a/policy/org.freedesktop.packagekit.policy.in
+++ b/policy/org.freedesktop.packagekit.policy.in
@@ -93,7 +93,7 @@
     <defaults>
       <allow_any>no</allow_any>
       <allow_inactive>no</allow_inactive>
-      <allow_active>yes</allow_active>
+      <allow_active>auth_admin_keep</allow_active>
     </defaults>
   </action>
 
@@ -131,7 +131,7 @@
     <defaults>
       <allow_any>no</allow_any>
       <allow_inactive>no</allow_inactive>
-      <allow_active>yes</allow_active>
+      <allow_active>auth_admin_keep</allow_active>
     </defaults>
   </action>
 
--- a/po/POTFILES.skip
+++ b/po/POTFILES.skip
@@ -6,4 +6,4 @@
 lib/packagekit-glib/egg-debug.c
 lib/packagekit-glib2/egg-debug.c
 backends/yum/libzif/egg-debug.c
-
+.pc/policy.diff/policy/org.freedesktop.packagekit.policy.in
